core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -552,6 +552,12 @@ public void testToQuery() throws IOException {
         }
     }
 
+    private QueryBuilder<?> rewriteQuery(QB queryBuilder, QueryRewriteContext rewriteContext) throws IOException {
+        QueryBuilder<?> rewritten = QueryBuilder.rewriteQuery(queryBuilder, rewriteContext);
+        assertSerialization(rewritten);
```

Comment:
will add a comment, this is basically testing the serialization of a rewritten query to ensure that the rewritten version is also serializable etc.